compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -697,7 +697,7 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
             // See doc comment for ElimByName for speculation how we could improve this.
         else
           MethodType(Nil, Nil,
-            eraseResult(sym.info.finalResultType.translateFromRepeated(toArray = sourceLanguage.isJava)))
+            eraseResult(rt.translateFromRepeated(toArray = sourceLanguage.isJava)))
```

Comment:
This was needed for [test `run/t3452g`](https://github.com/lampepfl/dotty/blob/master/tests/run/t3452g/A.scala)

The backend calls `TypeErasure.transformInfo(<Symbol TraversableLike.tail>, <Type ExprType(AbstractTrav[String])>)`.

Erasure should, I assume, continue to work on `AbstractTrav[String]` (which is `rt` here). Calling `sym.info.finalResultType` gives `Repr` which erases differently.

Reply:
Apparently that was intentionally changed in https://github.com/lampepfl/dotty/commit/76af6c53cd65a8569069b5fed00aa9ca12e30ee7

Reply:
Tests passed...

Reply:
Sure, but at the very least we need to carefully check if this breaks binary compatibility.

Reply:
Very true. How could I do that? Compile the community build twice and do a [jardiff](https://github.com/scala/jardiff)?

Reply:
I ran `dotty.communitybuild.CommunityBuildTestA` twice; once with this PR, once with the line here reverted. This resulted in 14081 identical classfiles.

I also added a check to see when the old and new erasure are different and it only happened when called from the backend:

```diff
+          val a = eraseResult(rt.translateFromRepeated(toArray = sourceLanguage.isJava))
+          val b = eraseResult(sym.info.finalResultType.translateFromRepeated(toArray = sourceLanguage.isJava))
+          if (!(a =:= b) && !(new Exception).getStackTrace.exists(_.toString.contains("getStaticForwarderGenericSignature"))) then
+            throw new Exception(s"[ERASURE]\n- $a\n- $b\n- $tp\n- ${sym.showFullName}")
           MethodType(Nil, Nil,
-            eraseResult(rt.translateFromRepeated(toArray = sourceLanguage.isJava)))
+            a)
```

I ran the `sbt test` and `CommunityBuildTestA` with this in place and did't see the exception showing up.

@smarter any other ideas?

Reply:
Thanks, I've run a similar test on the whole CI and it also didn't show any situation where the two types differed, this makes sense since the tp passed to eraseInfo should always be the symbol info at that point (it's only called from transformInfo which is only called from Erasure#transform, which always uses the symbol info except for `java.lang.Object` which is treated specially), so LGTM :sparkler: